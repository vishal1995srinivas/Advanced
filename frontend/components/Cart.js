import React from 'react';
import CartStyle from './styles/CartStyles';
import Supreme from './styles/Supreme';
import SickButton from './styles/SickButton';
import CloseButton from './styles/CloseButton';

const Cart = () => (
	<CartStyle open={true}>
		<header>
			<CloseButton title="close">&times;</CloseButton>
			<Supreme>Your Cart</Supreme>
			<p>You have items in your Cart</p>
		</header>
		<footer>
			<p>$10</p>
			<SickButton>Checkout</SickButton>
		</footer>
	</CartStyle>
);

export default Cart;
