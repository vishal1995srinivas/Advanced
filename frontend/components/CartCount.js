import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Dot = styled.div`
	background: ${(props) => props.theme.red};
	color: white;
	border-radius: 50%;
	padding: 0.5rem;
	line-height: 2rem;
	min-width: 3rem;
	margin-left: 1rem;
	font-weight: 100;
	font-feature-settings: 'tnum';
	font-variant-numeric: tabular-nums; /* these two will adjust the width for numbers -> skinny nums 1 to fatty nos like 5*/
`;

const CartCount = ({ count }) => <Dot>{count}</Dot>;

export default CartCount;
