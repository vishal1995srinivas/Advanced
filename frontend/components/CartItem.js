import React from 'react';
import formatMoney from '../lib/formatMoney';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const CartItemStyles = styled.li``;
const CartItem = (props) => (
	<CartItemStyles>
		<img width="100" src={props.cartItem.item.image} alt={props.cartItem.item.name} />
	</CartItemStyles>
);

CartItem.propTypes = {
	cartItem: PropTypes.object.isRequired
};
export default CartItem;
