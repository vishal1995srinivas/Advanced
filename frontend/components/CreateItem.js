import React, { Component } from 'react';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import Form from './styles/Form';
import formatMoney from '../lib/formatMoney';
import Error from './ErrorMessage';

const CREATE_ITEM_MUTATION = gql`
	mutation CREATE_ITEM_MUTATION(
		$title: String!
		$description: String!
		$price: Int!
		$image: String
		$largeImage: String
	) {
		createItem(title: $title, description: $description, price: $price, image: $image, largeImage: $largeImage) {
			id
		}
	}
`;
class CreateItem extends Component {
	constructor(props) {
		super(props);

		this.state = {
			title: 'Shoes',
			description: 'I love this description',
			image: 'dog.png',
			largeImage: 'large_dog.png',
			price: 1000
		};
	}
	handleChange = (e) => {
		const { name, type, value } = e.target;
		const val = type === 'number' ? parseFloat(value) : value;
		this.setState({ [name]: val });
	};

	render() {
		return (
			<Mutation mutation={CREATE_ITEM_MUTATION} variables={this.state}>
				{(createItem, { loading, error }) => (
					<Form
						onSubmit={(e) => {
							e.preventDefault();
							console.log(this.state);
						}}
					>
						<Error error={error} />
						<fieldset disabled={loading} aria-busy={loading}>
							<label htmlFor="title">
								Title
								<input
									type="text"
									id="title"
									name="title"
									required
									placeholder="Title"
									value={this.state.title}
									onChange={this.handleChange}
								/>
							</label>
							<label htmlFor="price">
								Price
								<input
									type="number"
									id="price"
									name="price"
									required
									placeholder="Price"
									value={this.state.price}
									onChange={this.handleChange}
								/>
							</label>
							<label htmlFor="price">
								Description
								<input
									type="text"
									id="description"
									name="description"
									required
									placeholder="Enter a Description"
									value={this.state.description}
									onChange={this.handleChange}
								/>
							</label>
							<button type="submit">Submit</button>
						</fieldset>
					</Form>
				)}>
			</Mutation>
		);
	}
}
export default CreateItem;
