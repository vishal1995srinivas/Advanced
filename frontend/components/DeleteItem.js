import React, { Component } from 'react';

class DeleteItem extends Component {
	constructor(props) {
		super(props);

		this.state = {};
	}

	render() {
		return <div>{this.props.children}</div>;
	}
}

export default DeleteItem;
