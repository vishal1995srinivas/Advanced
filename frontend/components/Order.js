import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { format } from 'date-fns';
import Head from 'next/head';
import gql from 'graphql-tag';
import formatMoney from '../lib/formatMoney';
import Error from './ErrorMessage';
import OrderStyles from './styles/OrderStyles';

const SINGLE_ORDER_QUERY = gql`
	query SINGLE_ORDER_QUERY($id: ID!) {
		order(id: $id) {
			id
			charge
			total
			createdAt
			user {
				id
			}
			items {
				id
				title
				description
			}
		}
	}
`;

class Order extends Component {
	static propTypes = {
		id: PropTypes.string.isRequired
	};
	render() {
		return (
			<Query query={SINGLE_ORDER_QUERY} variables={{ id: this.props.id }}>
				{({ data, error, loading }) => {
					if (loading) return <p>Loading...</p>;
					if (error) return <Error error={error} />;
					return (
						<OrderStyles>
							<Head>
								<title> Sick-Fits -Order {Order.id}</title>
							</Head>
							<p>
								<span>Order Id: </span>
								<span>{this.props.id}</span>
							</p>
							<p>
								<span>Charge</span>
								<span>{Order.charge}</span>
							</p>
							<p>
								<span>Date</span>
								<span>{format(order.createdAt, 'MMMM d, YYYY h:mm a')}</span>
							</p>
						</OrderStyles>
					);
				}}
			</Query>
		);
	}
}

export default Order;
