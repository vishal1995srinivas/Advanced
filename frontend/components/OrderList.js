import React, { Component } from 'react';
import { Query } from 'react-apollo';
import { formatDistance } from 'date-fns';
import Link from 'next/link';
import styled from 'styled-components';
import gql from 'graphql-tag';
import formatMoney from '../lib/formatMoney';
import OrderItemStyles from './styles/OrderItemStyles';

const USER_ORDERS_QUERY = gql`
	query USER_ORDERS_QUERY {
		orders(orderBy: createdAt_DESC) {
			orders(orderBy: createdAt_DESC) {
				id
				total
				createdAt
				items {
					id
					title
					price
					description
					quantity
					image
				}
			}
		}
	}
`;

class OrderList extends Component {
	render() {
		return (
			<Query query={USER_ORDERS_QUERY}>
				{({ data: { orders }, loading, error }) => {
					if (loading) return <p>Loading...</p>;
					if (error) return <Error error={error} />;
					<p>You have {orders.length} orders</p>;
				}}
			</Query>
		);
	}
}

export default OrderList;
