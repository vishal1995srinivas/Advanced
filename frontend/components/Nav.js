import Link from 'next/link';
import {Mutation} from 'react-apollo';
import User from './User';
import Signout from './SignOut';
import {TOGGLE_CART} from '../components/Cart';

import NavStyles from './styles/NavStyles';
const Nav = () => (
	<User>
		{({ data: { me } }) => (
			<NavStyles>
				<Link href="/Items">
					<a>Shop</a>
				</Link>
				{me && (
					<>
						<Link href="/sell">
							<a>Sell</a>
						</Link>
						<Link href="/orders">
							<a>Orders</a>
						</Link>
						<Link href="/me">
							<a>Account</a>
						</Link>
						<Signout></Signout>
						<Mutation mutation={TOGGLE_CART}>
						{(toggleCart)=>(
						<button onClick={toggleCart}>My Cart</button>
						)}
						</Mutation>
					</>
				)}
				{!me && (
					<Link href="/signup">
						<a>SignIn</a>
					</Link>
				)}
			</NavStyles>
		)}
	</User>
);
export default Nav;
