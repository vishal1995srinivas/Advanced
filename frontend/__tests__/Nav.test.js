import { mount } from 'enzyme';
import wait from 'waait';
import Nav from '../components/Nav';
import { CURRENT_USER_QUERY } from '../components/User';
import { MockedProvider } from 'react-apollo/test-utils';
import { fakeUser } from '../lib/testUtils';
import toJSON from 'enzyme-to-json';

const notSignedInMocks = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: { data: { me: null } }
	}
];

const signedInMocks = [
	{
		request: { query: CURRENT_USER_QUERY },
		result: { data: { me: fakeUser() } }
	}
];
describe('<Nav/>', () => {
	it('renders a minimal nav when signed out', async () => {
		const wrapper = mount(
			<MockedProvider mocks={notSignedInMocks}>
				<Nav />
			</MockedProvider>
		);
		await wait();
		wrapper.update();
		//console.log(wrapper.debug());
		const nav = wrapper.find("ul[data-test='nav']");
		console.log(nav.debug());
		expect(toJSON(nav)).toMatchSnapshot();
	});
	it('renders a full nav when signed in', async () => {
		const wrapper = mount(
			<MockedProvider mocks={signedInMocks}>
				<Nav />
			</MockedProvider>
		);
		await wait();
		wrapper.update();
		//console.log(wrapper.debug());
		const nav = wrapper.find("ul[data-test='nav']");
		const count = nav.find('div.count');
		expect(toJSON(count)).toMatchSnapshot();
		//console.log(count.debug());
	});
});
