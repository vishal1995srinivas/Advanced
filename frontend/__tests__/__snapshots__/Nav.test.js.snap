// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav/> renders a minimal nav when signed out 1`] = `
<ul
  className="NavStyles-sc-11c0d2g-0 iTUtFG"
  data-test="nav"
>
  <Link
    href="/Items"
  >
    <a
      href="/Items"
      onClick={[Function]}
    >
      Shop
    </a>
  </Link>
  <Link
    href="/signup"
  >
    <a
      href="/signup"
      onClick={[Function]}
    >
      SignIn
    </a>
  </Link>
</ul>
`;
